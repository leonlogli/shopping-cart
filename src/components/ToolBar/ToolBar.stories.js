import React from 'react'
import { action } from '@storybook/addon-actions'

import ToolBar from '.'

export default {
  title: 'ToolBar',
  parameters: { options: { selectedPanel: 'storybook/actions/panel' } },
}

export const Default = () => <ToolBar title="Shopping Cart" />

export const withLeftIconAction = () => (
  <ToolBar onLeftIconClick={action('click')} title="Shopping Cart" />
)

export const withChildAndRight = () => (
  <ToolBar right={<span>Right</span>}>
    <span>Child</span>
  </ToolBar>
)

export const withLeftIcon = () => (
  <ToolBar leftIcon="back" title="Shopping Cart" />
)
