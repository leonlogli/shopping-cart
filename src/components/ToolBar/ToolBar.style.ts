import styled, { css } from 'styled-components'

import { Box } from '../Layout'
import { H6 } from '../Typography'

const RightContainer = styled.div.attrs(() => ({
  className: 'RightContainer',
}))`
  display: flex;
  align-items: center;
  flex-grow: 1;
  justify-content: flex-end;
  margin-left: 8px;
`

const Title = styled(H6).attrs(() => ({
  className: 'Title',
  noWrap: true,
}))``

const ToolBarRoot = styled(Box)`
  display: flex;
  align-items: center;

  ${({ theme }) => css`
    padding-left: ${theme.spacing(2)}px;
    padding-right: ${theme.spacing(2)}px;

    ${theme.breakpoints.up('sm')} {
      padding-left: ${theme.spacing(3)}px;
      padding-right: ${theme.spacing(3)}px;
    }
  `}

  > .MuiIconButton-root:first-child {
    margin-left: -12px;
    margin-right: 12px;

    ${({ theme }) => css`
      ${theme.breakpoints.up('sm')} {
        margin-right: ${theme.spacing(2)}px;
      }
    `}
  }
`

export { ToolBarRoot, RightContainer, Title }
export default ToolBarRoot
