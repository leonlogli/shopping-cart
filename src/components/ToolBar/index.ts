export * from './ToolBar'
export { default } from './ToolBar'
