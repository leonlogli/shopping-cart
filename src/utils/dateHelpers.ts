const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
