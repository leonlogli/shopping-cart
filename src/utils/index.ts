export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
