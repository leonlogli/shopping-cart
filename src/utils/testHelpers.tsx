import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
