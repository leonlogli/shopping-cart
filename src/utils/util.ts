const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
