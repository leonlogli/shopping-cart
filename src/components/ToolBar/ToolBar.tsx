import React, { ReactNode, ReactElement } from 'react'
import IconButton from '@material-ui/core/IconButton'
import ArrowBackIcon from '@material-ui/icons/ArrowBack'
import MenuIcon from '@material-ui/icons/Menu'
import CloseIcon from '@material-ui/icons/Close'

import { BoxProps } from '../Layout'

import { ToolBarRoot, RightContainer, Title } from './ToolBar.style'

export type ToolBarProps = Pick<BoxProps, 'id' | 'className' | 'component'> & {
  /**
   * Children of the toolbar. they are placed at the left container of the toolbar
   */
  children?: ReactNode
  /**
   * Right items of the toolbar
   */
  right?: ReactNode
  /**
   * Title of the toolbar
   */
  title?: string
  /**
   * Icon to show (typically on mobile) at the left of the toolbar
   * @default 'menu' if onLeftIconClick is specified, 'undefined' otherwise
   */
  leftIcon?: 'menu' | 'back' | 'close' | ReactElement
  /**
   * Callback fired on `predefined` left icon clicked. This will not be invoked custom icon
   */
  onLeftIconClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
}

/**
 * Container for items such as application title, navigation icon, and action items.
 */
const ToolBar = (props: ToolBarProps) => {
  const {
    title,
    right,
    onLeftIconClick,
    children,
    leftIcon = onLeftIconClick && 'menu',
    ...other
  } = props

  const isPredefinedIcon =
    leftIcon === 'menu' || leftIcon === 'close' || leftIcon === 'back'

  return (
    <ToolBarRoot fullWidth {...other}>
      {isPredefinedIcon && (
        <IconButton
          aria-label={leftIcon as string}
          color="inherit"
          onClick={onLeftIconClick}
        >
          {leftIcon === 'menu' && <MenuIcon />}
          {leftIcon === 'close' && <CloseIcon />}
          {leftIcon === 'back' && <ArrowBackIcon />}
        </IconButton>
      )}
      {!isPredefinedIcon && leftIcon && leftIcon}
      {title && <Title>{title}</Title>}
      {children}
      {right && <RightContainer>{right}</RightContainer>}
    </ToolBarRoot>
  )
}

export { ToolBar }
export default ToolBar
