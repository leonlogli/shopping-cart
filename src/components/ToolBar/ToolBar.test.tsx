import React from 'react'

import ToolBar from '.'

import { mountWithTheme } from '../../utils'

describe('<ToolBar />', () => {
  const wrapper = mountWithTheme(<ToolBar title="Recipoz" />)

  it('should render children', () => {
    expect(wrapper.find('h6').props().children).toBe('Recipoz')
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should set right content', () => {
    const toolbar = mountWithTheme(<ToolBar right={<span>right</span>} />)

    expect(toolbar.find('span').props().children).toBe('right')
  })

  it('should have left icon', () => {
    const toolbar = mountWithTheme(<ToolBar leftIcon="menu" />)

    expect(toolbar.find('svg')).toHaveLength(1)
  })
})
