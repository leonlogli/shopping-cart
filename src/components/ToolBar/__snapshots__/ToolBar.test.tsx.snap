// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolBar /> should match a snapshot 1`] = `
<ToolBar
  title="Shopping Cart"
>
  <Styled(Component)
    fullWidth={true}
  >
    <ForwardRef
      className="cytxAi"
      fullWidth={true}
    >
      <Styled(Styled(MuiBox))
        className="cytxAi"
        fullWidth={true}
      >
        <Styled(MuiBox)
          className="iQnKkE cytxAi"
          fullWidth={true}
        >
          <div
            className="MuiBox-root MuiBox-root-1 iQnKkE cytxAi"
            fullWidth={true}
          >
            <Styled(H6)>
              <H6
                className="Title"
                noWrap={true}
              >
                <Styled(WithStyles(ForwardRef(Typography)))
                  className="Title"
                  noWrap={true}
                  variant="h6"
                >
                  <WithStyles(ForwardRef(Typography))
                    className="Title"
                    noWrap={true}
                    variant="h6"
                  >
                    <ForwardRef(Typography)
                      className="Title"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      noWrap={true}
                      variant="h6"
                    >
                      <h6
                        className="MuiTypography-root Title MuiTypography-h6 MuiTypography-noWrap"
                      >
                        Shopping Cart
                      </h6>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                </Styled(WithStyles(ForwardRef(Typography)))>
              </H6>
            </Styled(H6)>
          </div>
        </Styled(MuiBox)>
      </Styled(Styled(MuiBox))>
    </ForwardRef>
  </Styled(Component)>
</ToolBar>
`;
